--- code/qcommon/q_shared.h	2011-05-09 11:19:42.256914597 -0700
+++ code/qcommon/q_shared.h	2011-05-14 12:24:25.461067611 -0700
@@ -32,10 +32,10 @@
 // q_shared.h -- included first by ALL program modules.
 // A user mod should never modify this file
 
-#define PRODUCT_NAME            "XreaL" // Case, Spaces allowed
-#define PRODUCT_NAME_UPPPER     "XreaL" // Case, No spaces
-#define PRODUCT_NAME_LOWER      "xreal" // No case, No spaces
-#define PRODUCT_VERSION         "0.8.2"
+#define PRODUCT_NAME            "Dretchstorm" // Case, Spaces allowed
+#define PRODUCT_NAME_UPPPER     "DStorm" // Case, No spaces
+#define PRODUCT_NAME_LOWER      "dstorm" // No case, No spaces
+#define PRODUCT_VERSION         "0.0.1"
 
 #define ENGINE_NAME             "XreaL Engine"
 #define ENGINE_VERSION          "0.8.2"
diff -r 1971d59ed367 code/common/cmdlib.c
--- code/common/cmdlib.c	Sun Apr 17 01:40:59 2011 -0700
+++ code/common/cmdlib.c	Sun Apr 17 23:39:59 2011 -0700
@@ -41,7 +41,7 @@
 #include <libc.h>
 #endif
 
-#define	BASEDIRNAME	"xreal"		// assumed to have a 2 or 3 following
+#define	BASEDIRNAME	"dretchstorm"		// assumed to have a 2 or 3 following
 #define PATHSEPERATOR   '/'
 
 #ifdef SAFE_MALLOC
--- code/qcommon/qcommon.h	2011-05-09 11:19:42.276914597 -0700
+++ code/qcommon/qcommon.h	2011-05-14 13:01:55.521068199 -0700
@@ -254,14 +254,10 @@
 // NOTE: that stuff only works with two digits protocols
 extern int      demo_protocols[];
 
-#ifndef MASTER_SERVER_NAME
-#define MASTER_SERVER_NAME		"master.xreal-project.net"
-#endif
+#define MASTER_SERVER_NAME		"master.dretchstorm.com"
 
 #ifndef STANDALONE
-#ifndef AUTHORIZE_SERVER_NAME
-#define	AUTHORIZE_SERVER_NAME	"authorize.quake3arena.com"
-#endif
+#define	AUTHORIZE_SERVER_NAME	"authorize.dretchstorm.com"
 #ifndef PORT_AUTHORIZE
 #define	PORT_AUTHORIZE		27952
 #endif
@@ -631,9 +627,9 @@
 #define	MAX_FILE_HANDLES	64
 
 #ifdef DEDICATED
-#	define Q3CONFIG_CFG "xreal_server.cfg"
+#	define Q3CONFIG_CFG PRODUCT_NAME_LOWER "_server.cfg"
 #else
-#	define Q3CONFIG_CFG "xreal.cfg"
+#	define Q3CONFIG_CFG PRODUCT_NAME_LOWER ".cfg"
 #endif
 
 qboolean        FS_Initialized(void);
