diff -r 62944c8abf18 code/renderer/tr_bsp.c
--- code/renderer/tr_bsp.c	Sat Apr 16 14:04:08 2011 -0700
+++ code/renderer/tr_bsp.c	Sun Apr 17 01:34:56 2011 -0700
@@ -915,7 +915,7 @@
 
 				if(!lightmapFiles || !numLightmaps)
 				{
-					lightmapFiles = ri.FS_ListFilteredFiles(mapName, ".tga", "/lm_*.png", &numLightmaps);
+					lightmapFiles = ri.FS_ListFilteredFiles(mapName, ".tga", "/lm_*.tga", &numLightmaps);
 
 					if(!lightmapFiles || !numLightmaps)
 					{
@@ -944,7 +944,7 @@
 
 			if(!lightmapFiles || !numLightmaps)
 			{
-				lightmapFiles = ri.FS_ListFilteredFiles(mapName, ".tga", "/lm_*.png", &numLightmaps);
+				lightmapFiles = ri.FS_ListFilteredFiles(mapName, ".tga", "/lm_*.tga", &numLightmaps);
 
 				if(!lightmapFiles || !numLightmaps)
 				{
