diff -r 4b03aa34fc07 code/client/cl_ui.c
--- code/client/cl_ui.c	Thu Apr 14 02:10:15 2011 -0400
+++ code/client/cl_ui.c	Fri Apr 15 01:43:10 2011 -0400
@@ -293,6 +293,72 @@
 }
 
 /*
+==================
+G_SanitiseHostName
+
+Remove non-alphanumeric characters, black characters, and leading spaces from a host name
+==================
+*/
+static void G_SanitiseHostName( char *string )
+{
+	qboolean firstChar  = qfalse;
+	qboolean isBlack    = qfalse;
+	qboolean isGoodChar = qtrue;
+	qboolean skipSpaces = qtrue;
+
+	char *reader = string;
+	char *writer = string;
+
+	char lastChar = '\0';
+
+	while( *reader )
+	{
+		// Ignore leading spaces
+		if( *reader == ' ' && ( skipSpaces == qtrue || lastChar == ' ' ) )
+			isGoodChar = qfalse;
+
+		// Ignore black coloured characters
+		if ( lastChar == '^' && ColorIndex(*reader) == 0 )
+			isBlack = qtrue;
+
+		if ( isBlack && *reader != '^' )
+			isGoodChar = qfalse;
+		else if ( isBlack && *reader == '^' )
+			isBlack = isGoodChar = qfalse;
+		else
+			isBlack = qfalse;
+
+		// Ignore non-alphanumeric characters
+		if ( !isprint( *reader ) )
+			isGoodChar = qfalse;
+		else
+			skipSpaces = qfalse;
+
+		// Determine the first visible character
+		if ( !firstChar && lastChar != '^' && *reader != '^' )
+		{
+			// Strip the first visible character if it's a space
+			if ( *reader == ' ' )
+				isGoodChar = qfalse;
+			else
+				firstChar = qtrue;
+		}
+
+		if ( isGoodChar == qtrue )
+		{
+			*writer = *reader;
+			writer++;
+		}
+
+		isGoodChar = qtrue;
+		lastChar = *reader;
+		reader++;
+	}
+
+	*writer = '\0';
+}
+
+/*
 ====================
 LAN_GetServerInfo
 ====================
@@ -328,6 +394,7 @@
 	if(server && buf)
 	{
 		buf[0] = '\0';
+		G_SanitiseHostName( server->hostName );
 		Info_SetValueForKey(info, "hostname", server->hostName);
 		Info_SetValueForKey(info, "mapname", server->mapName);
 		Info_SetValueForKey(info, "clients", va("%i", server->clients));
