--- code/client/cl_main.c	2011-03-08 17:46:46.293923012 -0700
+++ code/client/cl_main.c	2011-03-12 16:14:47.131596005 -0700
@@ -232,8 +232,10 @@
 		reason = "Speex not initialized";
 	else if(!cl_connectedToVoipServer)
 		reason = "Server doesn't support VoIP";
+/* XXX
 	else if(Cvar_VariableValue("g_gametype") == GT_SINGLE_PLAYER || Cvar_VariableValue("ui_singlePlayerActive"))
 		reason = "running in single-player mode";
+*/
 
 	if(reason != NULL)
 	{
--- code/client/cl_parse.c	2011-03-08 17:46:46.283923011 -0700
+++ code/client/cl_parse.c	2011-03-12 16:15:31.271596002 -0700
@@ -419,9 +419,11 @@
 	// in the future, (val) will be a protocol version string, so only
 	//  accept explicitly 1, not generally non-zero.
 	s = Info_ValueForKey(systemInfo, "sv_voip");
+/* XXX
 	if(Cvar_VariableValue("g_gametype") == GT_SINGLE_PLAYER || Cvar_VariableValue("ui_singlePlayerActive"))
 		cl_connectedToVoipServer = qfalse;
 	else
+*/
 		cl_connectedToVoipServer = (atoi(s) == 1);
 
 #endif
--- code/server/sv_ccmds.c	2011-03-08 17:46:42.783923013 -0700
+++ code/server/sv_ccmds.c	2011-03-12 16:11:51.811596002 -0700
@@ -193,7 +193,7 @@
 	cmd = Cmd_Argv(0);
 	if(Q_stricmpn(cmd, "sp", 2) == 0)
 	{
-		Cvar_SetValue("g_gametype", GT_SINGLE_PLAYER);
+		// XXX Cvar_SetValue("g_gametype", GT_SINGLE_PLAYER);
 		Cvar_SetValue("g_doWarmup", 0);
 		// may not set sv_maxclients directly, always set latched
 		Cvar_SetLatched("sv_maxclients", "8");
@@ -213,10 +213,12 @@
 			cheat = qfalse;
 			killBots = qfalse;
 		}
+/* XXX
 		if(sv_gametype->integer == GT_SINGLE_PLAYER)
 		{
 			Cvar_SetValue("g_gametype", GT_FFA);
 		}
+*/
 	}
 
 	// save the map name here cause on a map restart we reload the q3config.cfg
--- code/server/sv_client.c	2011-03-08 17:46:42.783923013 -0700
+++ code/server/sv_client.c	2011-03-12 16:12:16.451596006 -0700
@@ -59,11 +59,13 @@
 	const char     *clientChallenge = Cmd_Argv(1);
 	challenge_t    *challenge;
 
+/* XXX
 	// ignore if we are in single player
 	if(Cvar_VariableValue("g_gametype") == GT_SINGLE_PLAYER || Cvar_VariableValue("ui_singlePlayerActive"))
 	{
 		return;
 	}
+*/
 
 	oldest = 0;
 	oldestTime = 0x7fffffff;
--- code/server/sv_main.c	2011-03-08 17:46:42.793923014 -0700
+++ code/server/sv_main.c	2011-03-12 16:12:43.491596001 -0700
@@ -608,11 +608,13 @@
 	char            infostring[MAX_INFO_STRING];
 	static leakyBucket_t bucket;
 
+/* XXX
 	// ignore if we are in single player
 	if(Cvar_VariableValue("g_gametype") == GT_SINGLE_PLAYER)
 	{
 		return;
 	}
+*/
 
 	// Prevent using getstatus as an amplifier
 	if(SVC_RateLimitAddress(from, 10, 1000))
@@ -672,11 +674,13 @@
 	char           *gamedir;
 	char            infostring[MAX_INFO_STRING];
 
+/* XXX
 	// ignore if we are in single player
 	if(Cvar_VariableValue("g_gametype") == GT_SINGLE_PLAYER || Cvar_VariableValue("ui_singlePlayerActive"))
 	{
 		return;
 	}
+*/
 
 	/*
 	 * Check whether Cmd_Argv(1) has a sane length. This was not done in the original Quake3 version which led
