diff -r 050d8bad1818 code/xmap2/game_xreal.h
--- code/xmap2/game_xreal.h	Sun Apr 17 01:37:57 2011 -0700
+++ code/xmap2/game_xreal.h	Sun Apr 17 01:40:10 2011 -0700
@@ -111,7 +111,7 @@
 {
 	"xreal",			/* -game x */
 	"base",				/* default base game data dir */
-	".xreal",			/* unix home sub-dir */
+	".dstorm",			/* unix home sub-dir */
 	"xreal",			/* magic path word */
 	"materials",		/* shader directory */
 	1024,				/* max lightmapped surface verts */
